### ISSUE TITLE
Alternative figure visualizations

### ISSUE THREAD

**Issue by willaguiar** [willaguiar/ASC_and_heat_transport#25]
We are looking into potential alternative ways to reformat the  $r^2$ vs $slope$ [figure](https://github.com/willaguiar/ASC_and_heat_transport/issues/20#issuecomment-1823850363) for easier visualization. This because the interpretation of the slope in this figure depends on which direction the ASC flows ( + or -), and if the CSHT is northward or southward.  Until now we came up with a few possibilities.

1-  Use transparency in the slope lines and r2 lines, whenever $r^2$  is too low, or the p value shows no significant correlation

2- Reformat each panel so negative slopes always means that as ASC strengthens the CSHT weakens, and positive slopes always means that as ASC  strengthens the CSHT also  strengthens .

3- Change the visualization to a scatter plot type, so the slope can be inferred visually, and add the depth information as a colorbar. 

I opened this issue so we can plot our test visualizations, and discuss whatever we find best.

**Comment by willaguiar** [comment]
Here is a visualization option 3, as suggested by @ongqingyee 
Bottom plots are for the same locations and timescale as the top one (Added for reference). I used transparency for $r^2$, so the more transparent the circle is, closer to 0 is the $r^2$ (check the legend in scatter on the left). Colors denote depth. I quite like this plot suggestion from Ellie because we can

I-  see the ASC speed and direction, and CSHT value and direction at the same time.
II- see that the $r^2$ are the best in deeper layers around the red colors, where the transparency is low.
III - See that at the surface, we have big ASC variability but quite low changes in CSHT (Horizontal perceived slope along the dark blue circles, check a,c). In turn in deep layers the perceived slope seems bigger.
IV- compare easily the magnitude of CSHT changes between regions (Same y-scale in all plots, but points extend further up and down on deep regions)

The only issue is that we don't have the information of the significance here (maybe we can find a way to fit in?)

<img width="910" alt="Screen Shot 2023-12-07 at 2 25 03 PM" src="https://github.com/willaguiar/ASC_and_heat_transport/assets/70033934/cbcb2289-0fdf-4b75-8ba3-5221f5f1be04">

![UvsCSHT_scatter](https://github.com/willaguiar/ASC_and_heat_transport/assets/70033934/a5ae6d6e-3fe9-4ab3-8ec8-190bfc401549)

**Comment by ongqingyee** [comment]
Looks cool, thanks! 

The negative regression slope at the deep layers with high r2 can be seen. But because of how much ASC variability there is compared to CSHT at shallower depth, there is less of an obvious trend. I guess if the paper focuses on CSHT and the ASC at depth this is less of an issue. 

It is good to see though, that at ~1000m for the highest r2 the same behaviour of less northward CSHT transport as ASC weakens still holds. Looks that the opposite is true at shallower depths, esp for the Deep regime?

I think we might have to eliminate depth levels that are not significantly correlated?

**Comment by taimoorsohail** [comment]
> I think we might have to eliminate depth levels that are not significantly correlated?

Agreed - maybe exclude points where R2 <0.5 AND  p-value >0.05? That will remove statistically insignificant correlations.

**Comment by wghuneke** [comment]
Here a got for option 2:

- Values of r2 with p-value <0.05 are removed (could update this to only show r2>0.5)
- the purple line is the new attempt of showing the slope: (i) the magnitude is the magnitude of the slope, (ii) the sign is positive when the CSHT increases for a stronger ASC and vice versa, (iii) the sign is negative when the CSHT decreases for a stronger ASC, (iv) the dashed line indicates southward CSHT and when the purple line is solid, there is northward heat transport.

The values are for monthly data, but I did a regime and time mean to do the final calculation to get the sign etc. 

![Fig_CSHT_vs_u_along_monthly_test2](https://github.com/willaguiar/ASC_and_heat_transport/assets/15355753/e002b248-7810-46d0-996f-229c3d5f96ba)

**Comment by willaguiar** [comment]
Good point Ellie/Taimoor

Here is an updated version of the fig, only for points where $r^2>0.5$, and $p<0.05$.The slopes and points groupings are very visible ( we can even add slope lines for the clusters we see). ps: everything in the reverse regime disappears ( no $r^2>0.5$). 
![UvsCSHT_scatter-2](https://github.com/willaguiar/ASC_and_heat_transport/assets/70033934/bbcd0bf5-fe33-4082-b2fe-5f40ff1fd2a5)

**Comment by willaguiar** [comment]
We had some suggestions that it would be interesting to see how the scatter plots would look like for the 9 panels with regime and timescales. Here it is ( using transparency for $r^2$)
![image](https://github.com/willaguiar/ASC_and_heat_transport/assets/70033934/ca7a4853-60e2-4774-b1cd-a88cd9307728)

based on the above plots, and if we filter $r^2<0.5$ and only regions with significant correlation, we can stack the plots my regime and get something like the fig below to append to the fig 2 of the manuscript
![image](https://github.com/willaguiar/ASC_and_heat_transport/assets/70033934/62059b1e-3972-400c-a79d-92c6b7a16fcb)
I kind of like the last one, as we can directly see that the biggest CSHT variability is in the deep regime, and compare the magnitude of changes in both ASC and CSHT between regimes and timescale quite easily. for a 9-plot version of this last figure check [the code line 157](https://github.com/willaguiar/ASC_and_heat_transport/blob/taimoor/Jupyter/GMM_ASC_newfigs.ipynb)
